src/core/common/settings.hpp

```diff
@@ -28,94 +28,430 @@
+        uint16_t        mRloc16;     ///< RLOC16
+        uint8_t         mMode;       ///< The MLE device mode
+    };
+
+    /**
+     * This constructor initializes the `Settings` object.
+     *
+     * @param[in]  aInstance     A reference to the OpenThread instance.
+     *
+     */
+    explicit Settings(Instance &aInstance)
+        : InstanceLocator(aInstance)
+    {
+    }
+
+    /**
+     * This method initializes the platform settings (non-volatile) module.
+     *
+     * This should be called before any other method from this class.
+     *
+     */
+    void Init(void);
+
+    /**
+     * This method removes all settings from the non-volatile store.
+     *
+     */
+    void Wipe(void);
+
+    /**
+     * This method saves the Operational Dataset (active or pending).
+     *
+     * @param[in]   aIsActive   Indicates whether Dataset is active or pending.
+     * @param[in]   aBuffer     A byte buffer containing the dataset.
+     * @param[in]   aSize       The size of buffer (number of bytes).
+     *
+     * @retval OT_ERROR_NONE              Successfully saved Dataset.
+     * @retval OT_ERROR_NOT_IMPLEMENTED   The platform does not implement settings functionality.
+     *
+     */
+    otError SaveDataset(bool aIsActive, const uint8_t *aBuffer, uint16_t aSize)
+    {
+        return Save(aIsActive ? kKeyActiveDataset : kKeyPendingDataset, aBuffer, aSize);
+    }
+
+    /**
+     * This method reads the Operational Dataset (active or pending).
+     *
+     * @param[in]   aIsActive             Indicates whether Dataset is active or pending.
+     * @param[out]  aBuffer               A byte buffer to output the read Dataset.
+     * @param[in]   aMaxBufferSize        Maximum size of buffer (number of bytes).
+     * @param[out]  aReadSize             A reference to a `uint16` to output the number of bytes read.
+     *
+     * @retval OT_ERROR_NONE              Successfully read the Dataset.
+     * @retval OT_ERROR_NOT_FOUND         No corresponding value in the setting store.
+     * @retval OT_ERROR_NOT_IMPLEMENTED   The platform does not implement settings functionality.
+     *
+     */
+    otError ReadDataset(bool aIsActive, uint8_t *aBuffer, uint16_t aMaxBufferSize, uint16_t &aReadSize)
+    {
+        return Read(aIsActive ? kKeyActiveDataset : kKeyPendingDataset, aBuffer, aMaxBufferSize, aReadSize);
+    }
+
+    /**
+     * This method indicates whether there is a previously saved Dataset (active or pending).
+     *
+     * @param[in]   aIsActive             Indicates whether Dataset is active or not.
+     *
+     * @retval OT_ERROR_NONE              There is a saved value in the settings.
+     * @retval OT_ERROR_NOT_FOUND         No corresponding value in the settings.
+     * @retval OT_ERROR_NOT_IMPLEMENTED   The platform does not implement settings functionality.
+     *
+     */
+    otError HasDataset(bool aIsActive) { return Has(aIsActive ? kKeyActiveDataset : kKeyPendingDataset); }
```

Comment:
Are these `Has*` methods used anywhere?

Reply:
Good question! :).... They are not used anywhere (I think we used this for `Dataset` but it's not needed anymore)... Removed them all in new pushed commit.